src/web/mod.rs

```diff
@@ -168,3 +159,38 @@ pub fn flash_redirect(msg: &str, path: &str, flash: Flash) -> ErrorResponse {
         .into_response()
         .into()
 }
+
+pub fn cached_asset<S>(
+    content: &'static [u8],
+    content_type: &'static str,
+) -> MethodRouter<S, Infallible>
+where
+    S: Clone + Send + Sync + 'static,
+{
+    let mut hasher = Md5::new();
+    hasher.update(content);
+
+    let content_hash = format!("{:x}", hasher.finalize());
+    get(move |request: axum::extract::Request| async move {
+        if let Some(header_value) = request.headers().get(axum::http::header::IF_NONE_MATCH) {
+            if header_value.to_str().unwrap_or("").eq(&content_hash) {
+                return Response::builder()
```

Comment:
`IntoResponse` is implemented for `StatusCode` so you can just do `StatusCode::NOT_MODIFIED.into_response()`
https://docs.rs/axum/latest/axum/response/trait.IntoResponse.html#impl-IntoResponse-for-StatusCode